sources/packages/backend/libs/integrations/src/esdc-integration/e-cert-integration/e-cert-file-handler.ts

```diff
@@ -274,115 +273,229 @@ export abstract class ECertFileHandler extends ESDCFileHandler {
 
   /**
    * Download all files from E-Cert Response folder on SFTP and process them all.
-   * @param eCertIntegrationService
+   * @param processSummary process summary of all files processed.
+   * @param eCertIntegrationService Integration service to read and delete the files.
    * @param fileCode ECert response file code to be processed.
-   * @returns Summary with what was processed and the list of all errors, if any.
+   * @param offeringIntensity offering intensity.
    */
   async processResponses(
+    processSummary: ProcessSummary,
     eCertIntegrationService: ECertIntegrationService,
     fileCode: string,
-  ): Promise<ProcessSFTPResponseResult[]> {
+    offeringIntensity: OfferingIntensity,
+  ): Promise<void> {
     const filePaths = await eCertIntegrationService.getResponseFilesFullPath(
       this.esdcConfig.ftpResponseFolder,
       new RegExp(`^${this.esdcConfig.environmentCode}${fileCode}`, "i"),
     );
-    const processFiles: ProcessSFTPResponseResult[] = [];
+    // Return if there are no files to be processed.
+    if (!filePaths.length) {
+      processSummary.info(
+        `There are no disbursement feedback error files to be processed for ${offeringIntensity}.`,
+      );
+    }
+    // Get eCert feedback error map for all the error codes.
+    let eCertFeedbackErrorCodeMap: Record<string, number>;
```

Comment:
As a suggestion, `Record<string, number>` is used multiple times and it can have a type created just to improve the readability.
```ts
type ECertFeedbackCodeMap = Record<string, number>;
```